### DISCUSSION TITLE
Tubular RC1

### DISCUSSION THREAD

**Discussion by polymorphicshade** [polymorphicshade/NewPipe#260] [Announcements]
# Tubular RC1
Here is an early release-candidate of Tubular:
Release APK (signed by me): [tubular_0.0.1_rc1.zip](https://github.com/polymorphicshade/NewPipe/files/9816493/tubular_0.0.1_rc1.zip)
# Features
The following features are specific to Tubular (when compared to vanilla NewPipe):
- new icon
- hide videos containing keywords and/or regex
- hide "YouTube Shorts"
- view comment replies
- YouTube account support
  - import subscriptions
  - view (a.k.a "explore") YouTube video suggestions
  - let YouTube track requests using your cookies to potentially affect your video suggestions
- SponsorBlock support
  - view/skip marked segments
  - submit segments
  - vote on segments (tap and hold a thumb to reset your vote)
  - whitelist channels
  - customize API endpoint
- ReturnYouTubeDislike (RYD) support
  - view dislikes on a video
  - customize API endpoint
# Why did this take so long?
I'm a busy fella and this project is low on my priority list. 😊
# Looking for feedback
Please use this discussion thread to let me know about any bugs and other feedback.
There is still more work to do on Tubular, but I wanted to get this RC out with the core changes I've been focusing on.
# YouTube account support? I'm concerned about my privacy...
As long as you don't sign-in to your YouTube account, Tubular will behave exactly like vanialla NewPipe does when sending web requests.
If you signed in but want to restore your privacy, go to the Account settings category and clear your sign-in cookies.

**Comment by catalin560** [comment]
wait, I'm confused... is this still part of NewPipe or is it your own spin?

**Reply by Midou36O** [reply]
It's based on newpipe from what i see, but wether the project is going to be developed in another repo or will replace this repo is another question...

**Reply by Midou36O** [reply]
From the RC i believe tubular will replace newpipe X sponsorblock.

**Reply by polymorphicshade** [reply]
I forgot to mention that!

Yes Tubular will replace this fork eventually. Once I'm comfortable with the code base, I'll rename this repo and delete my "sponsorblock" branch.

**Reply by Midou36O** [reply]
Looks good! I wonder if that recent change will allow you to get more frquent newpipe updates (if you cannot do it due to lack of time, allow someone else that is trustworthy to manage the repo while keeping control of it)

**Reply by polymorphicshade** [reply]
Not exactly sure what you mean, but generally keeping my fork up to date with upstream NewPipe is very simple, and usually takes me an hour or two.

There isn't all much difference between how I update the sponsorblock branch with the tubular one.

**Reply by Midou36O** [reply]
I meant, assuming tubular is going to slowly stop following newpipe's philosophy, my question is will there be more frequent updates that does not rely on newpipe's ones? (Sorry if that wasn't worded correctly.)

**Reply by polymorphicshade** [reply]
Oh no I still want this fork to basically be NewPipe, but with some additional optional features. I'm not trying to turn this into it's own little world.

I still want to keep it in line with upstream NewPipe.

The only philosophical differences we have involve advertising and censorship.

**Comment by MF-Debug** [comment]
This is revolutionary! Finally I can ditch (will wait a bit more to test) any modified youtube clients from my device! Working fine on my Samsung Galaxy Note 3 running LOS 18.1. Thank you so much!

**Comment by indrastorms** [comment]
![Screenshot_20221020-003759_Tubular RC1](https://user-images.githubusercontent.com/48489074/196784376-0475a04a-4a22-4500-9c23-15202d608018.png)

Some padding issue, the submit button is getting hidden. Otherwise great update.

**Reply by polymorphicshade** [reply]
Nice, guess I gotta check more screen sizes 😅

**Comment by pokegamer5547** [comment]
Hi! I tested this release once you released for few days and so far it looks so great! :) filtering is a lifesaver. I noticed few issues tho.

1. When hiding videos of certain uploaders,it doesnt hide the "channels" from search results itself. Which is bit weird since after restarting the app it wont really show up their videos even when you click on their channel. It would be great if it could hide channel from results too which we choosed to hide. 

2. Maybe adding a block button on channels page would make hiding certain channels way easier.


3. This release is crashing a lot for me tbh but maybe it's a upstream issue? I haven't tested upstream release yet.

4. Advance filtering (with java syntax) is really confusing for me tbh. I know how to use regex but this one is just not working at all.

**Reply by polymorphicshade** [reply]
Glad you think it's mostly good!

1. I did notice this, but I want to hold off on figuring it out, since my filtering code only works on video responses coming back from the Extractor. Making it also hide channels will significantly increase the scope of the code changes.
2. Channel-blocking has the same issue - my filtering code only works for videos. Nonetheless channel blocking is a great idea. I think you should make a feature request for it.
3. 99% of the crash emails I see are from YouTube issues or extractor issues (before my tubular branch of the extractor). Can you post the most frequent crashes here? The stack trace will help me determine if it's something I caused in my tubular branch.
4. Advance filtering was just thrown in there without much thought on the UX. It's meant more for users that want to fine-tune their filtering. I honestly can't help much with that since I really suck at regex...
But I do know of some simple regex you could try:
`.*                  -> show all videos`
`^(?:(?!word).)*$    -> hide videos that DON'T contain "word"`
`^.*word.*$          -> hide videos that contain "word"`

**Reply by Midou36O** [reply]
> I honestly can't help much with that since I really suck at regex...

This may help you in the future:
https://github.com/ziishaned/learn-regex/blob/master/README.md

**Comment by TheRealConfused** [comment]
Lovely! 


Is it possible to lock the video player into place when scrolling down to read comments much like YouTube? LibreTube also locks the video player into place when scrolling down to read the comments.

**Reply by polymorphicshade** [reply]
It is technically possible, but the UI was built in such a way that it would require a lot of UI re-writes. This would make keeping this fork in line with NewPipe much more difficult and time-consuming.

Best thing to do is to keep pressuring the NewPipe devs for this change.

**Comment by jivemelon** [comment]
Will there be an update on this project anytime soon? Seems like a few things are broken or misbehaving such as the What's New feed not loading new items and channel pages loading blank (although I'm aware that this is the case for Newpipe as well.) Otherwise great first start and I'm excited to see the development of this!

**Reply by polymorphicshade** [reply]
You are describing issues already being worked on by the vanilla NewPipe devs.

Tubular will be updated after the release of a new version of vanilla NewPipe.

**Reply by TDavLinguist** [reply]
If you know how to compile, you can just download the source from the :sponsorblock branch rather than the default :tubular branch and then set the extractor commit ID in the ./app/build.gradle file to `com.github.TeamNewPipe:NewPipeExtractor:eb07d70a2ce03bee3cc74fc33b2e4173e1c21436` and it should build the previous 0.24.0 NewPipeSponsorBlock release with the fixed Extractor. I did it and it seems to work fine for me. I'll include the ZIP here if you'd like to try it, but don't expect it to be bug free. 
[NPSB-newextractor.zip](https://github.com/polymorphicshade/NewPipe/files/9945928/NPSB-newextractor.zip)

**Comment by yoshimo** [comment]
will you expand account support to optionally  comment on content and take part in live chats and only send tracking information if you use either?

**Reply by polymorphicshade** [reply]
I have no plans on implementing live chat, but I do want to implement leaving comments on videos.
That probably won't come for a while though, not sure how I would do that yet.

**Comment by ghost** [comment]
Are there any plans to implement playlist sync? Looking forward to Tubular, loved the RC. Awesome job 😄

**Reply by polymorphicshade** [reply]
Actually yes! It's on my list of stuff to do.

**Comment by toas-koas** [comment]
It connects the domains
googleads.g.doubleclick.net
static.doubleclick.net

When I add an account

Is this necessary?

**Reply by polymorphicshade** [reply]
All signing in does is open a web browser to Google's sign in page, then when the page closes it captures the cookies generated by the sign in, and stores them for use later.

Whatever redirect the web browser goes to after signing in is up to the redirect.

So I'm not totally sure its necessary, but I don't know how I could stop a redirect through the web browser.

**Comment by Dragonfly-30** [comment]
Is it still being worked on? No update for more than 5 months.

**Reply by polymorphicshade** [reply]
Not actively worked on at the moment no, but it's not dead. It's the lowest priority in my list of current projects.

**Comment by DI555** [comment]
@polymorphicshade , great work!
could you pls, create a personal repo for Tubular, and just do updating the NP extractor core   - imo at least in a rare time when NewPipe updating it!
so, that way it’ll be usable all the time, but being still in the developement thou..
Many thanks in advance!
